### PR TITLE
Change one GitHub Workflow CI check to use C++20

### PR DESCRIPTION
**Pull request by masterleinad**
In response to https://github.com/dealii/dealii/pull/13508#issuecomment-1064742478.

### CODE REVIEW

**Review by peterrum — APPROVED** [review]
Thanks!

**Review by drwells — CHANGES_REQUESTED** [review]

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -24,10 +24,18 @@ jobs:
         cmake --version
     - name: modules
       run: |
-        sudo apt-get install libboost-all-dev
+        sudo apt-get install g++-10
+    - name: Build Boost
+      id: boost
+      uses: egor-tensin/build-boost@v1
```
[the remark this answers is not in the document]
Comment (masterleinad) [reply .github/workflows/linux.yml:30]: I ran into a compilation problem with C++20 that was resolved in later versions.

**Review by drwells — APPROVED** [review]

File: .github/workflows/linux.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (luca-heltai) [reply .github/workflows/linux.yml:30]: @masterleinad, out of curiosity, was the compilation problem related to the python bindings, or something more essential?

[the remark this answers is not in the document]
Comment (masterleinad) [reply .github/workflows/linux.yml:30]: It was something more essential. There was some kind of incompatibility of older boost versions with newer gcc versions and C++20 IIRC. I don't remember the specific boost library but I'm pretty sure it wasn't python.

**Pull request merged**